internal/pkg/mining/scheduler.go

```diff
@@ -54,16 +67,17 @@ type timingScheduler struct {
 // It returns a channel for reading mining outputs and a waitgroup for teardown.
 // It is the callers responsibility to close the out channel only after waiting
 // on the waitgroup.
-func (s *timingScheduler) Start(miningCtx context.Context) (<-chan Output, *sync.WaitGroup) {
+func (s *timingScheduler) Start(miningCtx context.Context) (<-chan FullBlock, *sync.WaitGroup) {
```

Comment:
I think the channel returned here can be replaced by a callback function/handler passed to `Start` (follow-up).

Reply:
It is true that this channel could be replaced with a callback, but I disagree that it would be preferable. this channel is used to communicate multiple values from goroutine. It's as good a use for a channel as I can imagine. A callback will need to replicate the close behavior and probably introduce locking. I'm strongly in favor of reducing complexity where we can, but switching to a callback seems a lateral move at best and is less idiomatic. If you have reasons why callbacks would be superior in these situations, we should discuss. It represents a shift in pattern that either should be replicated across the code base or left alone.

Reply:
Good points. I wish for a mechanism where a new block emitted by our miner can be processed by the chain sync system synchronously before the miner then starts the next cycle, so it can deterministically build on top of that block. Especially for testing. In production maybe we would have the callback just plop the block on a channel, but the point is that the caller gets to decide whether to introduce async.